funcx_sdk/funcx/utils/url_parsing.py

```diff
@@ -0,0 +1,10 @@
+from urllib.parse import urlparse
+
+
+def ws_uri_from_service_address(service_address):
+    url_data = urlparse(service_address)
+    scheme = "wss" if url_data.scheme == "https" else "ws"
+    hostname = url_data.hostname
+    port_str = "" if url_data.port is None else f":{url_data.port}"
```

Comment:
Should I make this more testing friendly? Like if it sees port 5000, set the ws_uri port to 6000 since this is what we use in testing